internal/backend_s3.go

```diff
@@ -667,7 +667,7 @@ func (s *S3Backend) CopyBlob(param *CopyBlobInput) (*CopyBlobOutput, error) {
 
 	params := &s3.CopyObjectInput{
 		Bucket:            &s.bucket,
-		CopySource:        aws.String(pathEscape(from)),
+		CopySource:        aws.String(url.QueryEscape(from)),
```

Comment:
Use the `QueryEscape` to encode URLEncode the params.  The `from` is a leaf node (just name, not a path) hence `QueryEscape` will escape all special characters.  Even if the `from` has a forward slash (`/`) then it will be escaped and AWS unescaping and working correctly.